tools/downloader/src/open_model_zoo/model_tools/converter.py

```diff
@@ -104,103 +104,131 @@ def main():
-        if unknown_precisions:
-            sys.exit('Unknown precisions specified: {}.'.format(', '.join(sorted(unknown_precisions))))
-
-    models = _common.load_models_from_args(parser, args)
-
-    output_dir = args.download_dir if args.output_dir is None else args.output_dir
-
-    def convert(reporter, model):
-        if model.mo_args is None:
-            reporter.print_section_heading('Skipping {} (no conversions defined)', model.name)
-            reporter.print()
-            return True
-
-        model_precisions = requested_precisions & model.precisions
-        if not model_precisions:
-            reporter.print_section_heading('Skipping {} (all conversions skipped)', model.name)
-            reporter.print()
-            return True
-
-        (output_dir / model.subdirectory).mkdir(parents=True, exist_ok=True)
-
-        if not run_pre_convert(reporter, model, output_dir, args):
-            return False
-
-        model_format = model.framework
-
-        template_variables = {
-            'config_dir': _common.MODEL_ROOT / model.subdirectory,
-            'conv_dir': output_dir / model.subdirectory,
-            'dl_dir': args.download_dir / model.subdirectory,
-            'mo_dir': mo_path.parent,
-        }
-
-        if model.conversion_to_onnx_args:
-            if not convert_to_onnx(reporter, model, output_dir, args, template_variables):
+    try:
+        telemetry = _common.start_telemetry('Model Converter')
+
+        if args.all:
+            telemetry.send_event('md', 'converter_selection_mode', 'all')
+        if args.list:
+            telemetry.send_event('md', 'converter_selection_mode', 'list')
+        if args.name:
+            telemetry.send_event('md', 'converter_selection_mode', 'name')
+
+        models = _common.load_models_from_args(parser, args)
+        for model in models:
+            telemetry.send_event('md', 'converter_model_name', model.name)
+            telemetry.send_event('md', 'converter_framework', model.framework)
+
+        mo_path = args.mo
+        if mo_path is None:
+            try:
+                mo_path = Path(os.environ['INTEL_OPENVINO_DIR']) / 'deployment_tools/model_optimizer/mo.py'
+            except KeyError:
+                telemetry.send_event('md', 'converter_error', str({'error': 'unable-to-locate-mo', 'model': 'none'}))
+                sys.exit('Unable to locate Model Optimizer. '
+                    + 'Use --mo or run setupvars.sh/setupvars.bat from the OpenVINO toolkit.')
+
+        extra_mo_args = args.extra_mo_args or []
+
+        if args.precisions is None:
+            requested_precisions = _common.KNOWN_PRECISIONS
+        else:
+            requested_precisions = set(args.precisions.split(','))
+            unknown_precisions = requested_precisions - _common.KNOWN_PRECISIONS
+            if unknown_precisions:
+                telemetry.send_event('md', 'converter_error', str({'error': 'unknown-precisions', 'model': 'none'}))
```

Comment:
Is this error worth reporting? It's basically just an invalid argument value - I'm not sure we're gonna get any insights by collecting these.

Reply:
Ok, removed this event. Do we need then `unable-to-locate-mo/pot` error reporting?

Reply:
I'm a bit torn on those, but I think they could be useful - they indicate how often the users try to run our tools without setting up the OpenVINO environment.

Reply:
On second thought, I don't think the telemetry library can even be loaded unless the OpenVINO environment is set up, so maybe those events are pointless too.